app/controllers/comment.js

```diff
@@ -0,0 +1,131 @@
+    console.log(id)
+    try {
+      const comment = await CommentModel.findByIdAndRemove(id)
+      if (!comment) {
+        return res.status(400).json({ error: 'No comment exixts' })
+      }
+      res.status(200).json({ comment: comment })
+    } catch (error) {
+      console.log(error)
+      res.status(400).json({ error: error })
+    }
+  },
+  update: async (req, res, next) => {
+    const { id } = req.params
+    const updates = Object.keys(req.body)
+    let isValidUpdate = true
+    if (updates !== 'content') {
+      isValidUpdate = false
+    }
+    if (!isValidUpdate) {
+      return res.status(400).json({ error: 'Wrong Update Request' })
+    }
+    try {
+      const comment = await CommentModel.findById(id)
+      updates.forEach(update => {
+        comment[update] = req.body[update]
+      })
+      await comment.save()
+      res.status(204).json({ comment: comment })
+    } catch (error) {
+      console.log(error)
+      res.status(400).json({ error: error })
+    }
+  },
+  getCommentByPost: async (req, res, next) => {
+    const { id } = req.params
+    try {
+      await CommentModel.find({ postId: id }, (error, comments) => {
+        if (error) {
+          return res.status(400).json({ error: error })
+        }
+        if (!comments.length) {
+          return res.status(404).json({ message: 'Item not found' })
+        }
+        return res.status(200).json({ success: true, comments: comments })
+      })
+    } catch (error) {
+      console.log(error)
+      res.status(400).json({ error: error })
+    }
+  },
+  upvote: async (req, res, next) => {
+    const { id, userId } = req.params
+    try {
+      const comment = await CommentModel.findById(id)
+      if (!comment) {
+        return res.status(400).json({ error: 'No comment found' })
+      }
+      const noUpvotes = comment.votes.upVotes.user.user_id.length
+      // CHECKS IF THE USER HAS ALREADY UPVOTED THE COMMENT
+      let i
+      for (i = 0; i < noUpvotes; i++) {
+        if (JSON.stringify(comment.votes.upVotes.user.user_id[i]) === JSON.stringify(userId)) {
+          return res.status(400).json({ error: 'Bad request' })
+        }
+      }
+      // CHECKS IF THE USER HAS ALREADY DOWNVOTED THE COMMENT
+      const noDownvotes = comment.votes.downVotes.user.user_id.length
+      for (i = 0; i < noDownvotes; i++) {
+        if (JSON.stringify(comment.votes.downVotes.user.user_id[i]) === JSON.stringify(userId)) {
+          comment.votes.downVotes.count = comment.votes.downVotes.count - 1
+          comment.votes.downVotes.user.user_id.remove(comment.votes.downVotes.user.user_id[i])
+        }
+      }
+      comment.votes.upVotes.count = comment.votes.upVotes.count + 1
+      comment.votes.upVotes.user.user_id.push(userId)
+      comment.save()
+      res.status(200).json({ comment: comment, message: 'Success' })
+    } catch (error) {
+      console.log(error)
+      res.status(400).json({ error: error })
+    }
+  },
```

Comment:
@rak-shit Instead of making two api for upvote and downvote
you should make a single API that would be working for both upvote and downvote
saves a lot of code

Reply:
Can you elaborate how would that decrease the number of lines.

Reply:
send boolean values for upvote and downvote 
now match which one is true (you can use if else) and manipulate the code accordingly in favour of the one which is true

Reply:
Yeah I got what you are saying but I feel it would hardly decrease the number of lines of code. Moreover, I did it this way because it would be easy to perceive for the reader. Though I will try doing what you suggested, if that makes a significant difference in the number of lines I will adopt it. :)